### ISSUE TITLE
Unable to use external libraries via 'using'

### ISSUE THREAD

**Issue by bc3tech** [lukebuehler/CShell#16]
1. Create simple external library (DLL)
2. Add the lib as a reference to the CShell workspace
3. Add a 'using ___' statement to Scratchpad.csx for the namespace of the lib
4. Notice no commands following the 'using' statement execute correctly:

```
using ClassLibrary1;

var c = new Class1();
c.SomeMethod();
```

```
[Evaluating external code (Scratchpad.csx)]
(7,0): error CS1525: Unexpected symbol `var'
```

![classlibrary1 dll](https://cloud.githubusercontent.com/assets/1967433/2997943/1f92f054-dd05-11e3-8912-53b44c45e84c.gif)

I looked around the site, wiki, and Tutorial.csx and couldn't find any guidance on using external libraries, so if I'm doing something wrong one way to resolve this issue would be to add some doco to that affect. :)

**Comment by lukebuehler** [comment]
Yep, using statements should be executed separately right now. I usually have a file called `usings.csx` and add the using statements there. But this needs to be fixed to work properly, especially with the upcoming switch to the Roslyn/ScriptCs engine.

**Comment by bc3tech** [comment]
Ah. Good tip. We should make this part of the current documentation/wiki.

B

Sent from my Windows Phone
